sdk/common/src/main/java/io/opentelemetry/sdk/common/export/ProxyOptions.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.common.export;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.Proxy;
+import java.net.ProxySelector;
+import java.net.SocketAddress;
+import java.net.URI;
+import java.util.Collections;
+import java.util.List;
+
+/** Configuration for proxy settings. */
+public final class ProxyOptions {
+  private final ProxySelector proxySelector;
+
+  private ProxyOptions(ProxySelector proxySelector) {
+    this.proxySelector = proxySelector;
+  }
+
+  /** Create proxy options with the {@code proxySelector}. */
+  public static ProxyOptions create(ProxySelector proxySelector) {
+    return new ProxyOptions(proxySelector);
+  }
+
+  /**
+   * Create proxy options with a {@link ProxySelector} which always uses an {@link Proxy.Type#HTTP}
+   * proxy with the {@code socketAddress}.
+   */
+  public static ProxyOptions create(InetSocketAddress socketAddress) {
+    return new ProxyOptions(new SimpleProxySelector(new Proxy(Proxy.Type.HTTP, socketAddress)));
+  }
+
+  /** Return the {@link ProxySelector}. */
+  public ProxySelector getProxySelector() {
+    return proxySelector;
+  }
+
+  @Override
+  public String toString() {
+    return "ProxyOptions{proxySelector=" + proxySelector + "}";
+  }
+
+  private static final class SimpleProxySelector extends ProxySelector {
+
+    private final List<Proxy> proxyList;
+
+    private SimpleProxySelector(Proxy proxy) {
+      this.proxyList = Collections.singletonList(proxy);
+    }
+
+    @Override
+    public List<Proxy> select(URI uri) {
+      return proxyList;
+    }
+
+    @Override
+    public void connectFailed(URI uri, SocketAddress sa, IOException e) {
+      // ignore
```

Comment:
Could it make sense to at least log out some debug logs here to indicate that there was a problem? I am not sure how common that is in the opentelemetry project to log such things though. I am not sure how verbose the client implementations are about proxy failures.

Reply:
FWIW, the static inner one in java.net just says `/* ignore */` as well. I wonder if the user is supposed to care....

Reply:
The purpose of this method seems to be to allow the selector to choose another proxy for subsequent requests in the case of error, i.e. for load balancing.